File: SearchingSortingAndPathfinding/Queue.py
class Node():
	def __init__(self, data, link):
		self.data = data
		self.link = link

	def __repr__(self):
		return "Node <{}>".format(self.data)

class Queue():

	def __init__(self):
		self.front, self.rear = None, None

	def Enqueue(self, data):
		if(self.front == None and self.rear == None):
			self.front = self.rear = Node(data, None)
			return

		self.rear.link = Node(data, None)
		self.rear = self.rear.link

	def Dequeue(self):
		if(self.front == None):
			print("Dequeue from empty queue.\n")
			return

		data = self.front.data
		self.front = self.front.link

		return data


	def isEmpty(self):
		return (self.front == None and self.rear == None)

queue = Queue()
for i in range(0, 9):
	queue.Enqueue(i)

for i in range(0, 5):
	print(queue.Dequeue())

File: StacksAndQueues/Queue.cs
using System.Diagnostics;

namespace DataStructures.StacksAndQueues
{
    public class Queue
    {
        //Set up a Node class
        public class Node
        {
            private object _data;
            private Node _next;

            public object Data { get { return this._data; } set { this._data = value; } }
            public Node Next { get { return this._next; } set { this._next = value; } }

            public Node(object data = null, Node next = null)
            {
                this._data = data;
                this._next = next;
            }
        }

        //Start Queue class
        private Node front;
        private Node rear;
        private int size;

        public Queue()
        {
            this.front = this.rear = null;
            this.size = 0;
        }

        public void Enqueue(object data)
        {
            //Make a new node to hold the data
            Node temp = new Node(data);

            /*If queue is empty, set this new node as both the front and rear of the queue
             *If not, just attach the new node to the rear of the queue*/
            if (this.IsEmpty())
            {
                this.front = this.rear = temp;
            }
            else
            {
                this.rear.Next = temp;
                this.rear = this.rear.Next;
            }

            //Increment the size of the queue
            this.size++;
            
        }

        public object Dequeue()
        {
            object value = null;

            //If queue is empty, don't dequeue anything and let the user know
            if (this.IsEmpty())
            {
                Debug.WriteLine("Nothing to Dequeue.");
            }
            else
            {
                value = this.front.Data;
                this.front = this.front.Next;
                this.size--;
            }

            return value;

        }

        public int GetSize()
        {
            return this.size;
        }

        public bool IsEmpty()
        {
            return this.size == 0;
        }

        public void Clear()
        {
            this.front = this.rear = null;
            this.size = 0;
        }

    }
}
